gene_transformer/model.py

```diff
@@ -173,10 +189,20 @@ def train(cfg: ModelSettings) -> None:
     # Setup wandb
     wandb_logger = None
     if cfg.wandb_active:
-        node_rank = os.environ.get("NODE_RANK", 0)
-        print(f"Node rank: {node_rank}")
-        if node_rank == 1:
-            print("Using Weights and Biases for logging...")
+        node_rank = os.environ.get("NODE_RANK")
+        rank = os.environ.get("RANK")
+        local_rank = os.environ.get("LOCAL_RANK")
+        slurm_procid = os.environ.get("SLURM_PROCID")
+        jsm_namespace = os.environ.get("JSM_NAMESPACE_RANK")
+        wandb_active_env = os.environ.get("WANDB_ACTIVE")
+
+        print(
+            f"{rank=}, {local_rank=}, {slurm_procid=}, {jsm_namespace=}, {node_rank=}"
```

Comment:
We probably want to get rid of this print and os.environ calls to keep the script clean and avoid needing to specify default arguments to .get() that may confuse. 

The new logic looks great!

Reply:
The if statement can simply be:

```
if os.environ.get("WANDB_ACTIVE", 1):
```

Reply:
This can be added to line 191